packages/react-reconciler/src/__tests__/ReactSuspenseWithNoopRenderer-test.js

```diff
@@ -623,6 +623,17 @@ describe('ReactSuspenseWithNoopRenderer', () => {
       // The new reconciler batches everything together, so it finishes without
       // suspending again.
       'Sibling',
+
+      // NOTE: The final of the update got pushed into a lower priority range of
+      // lanes, leading to the extra intermediate render. This is because when
+      // we schedule the fourth update, we're already in the middle of rendering
+      // the three others. Since three are only three lanes in the default
+      // range, the fourth lane is shifted to slightly lower priority. This
```

Comment:
The "lane priority" of a bit is determined by which range of bits it belongs to. We can't assign a new update to a lane that's already in the middle of rendering. In this (fairly contrived) test scenario, we ran out of lanes in the Default range so we had to pick from the next available lane in a lower priority range. Another option would be to throw out the in-progress work and start over — then we're free to pick any lane we want. tl;dr this is a batching heuristic that doesn't matter that much in most scenarios and probably isn't the last time this will change. It's only observable in tests if you do clever stuff like partial rendering, which we don't officially support and is only used in our own test suite.

Reply:
I think what's confusing to me in "the fourth lane is shifted to slightly lower priority" is that the _lane_ isn't what changes priorities. Do you mean "the fourth **update** is shifted to a (lane in a) slightly lower priority"?

Reply:
Oh right. Yeah I meant fourth update :D